test/IntegrationTest/carService.test.ts

```diff
@@ -0,0 +1,223 @@
+import 'dotenv/config'
+import { connection } from '../../src/Repositories/data-source';
+import { Car } from '../../src/Model/Vehicle';
+import * as carService from '../../src/Services/carService'
+import { Chassis } from '../../src/Model/Chassis';
+import { Engine } from '../../src/Model/Engine';
+import { FuelStats } from '../../src/Model/FuelStats';
+import { Gearbox } from '../../src/Model/Gearbox';
+import { Manufacturer } from '../../src/Model/Manufacturer';
+import { Registration } from '../../src/Model/Registration';
+
+let car: Car;
+let cars: Car[];
+let choice: number;
+
+
+/* ---------------------------------------- generatecar ---------------------------------------- */
+describe('generatecar', () => {
+
+    beforeEach(async () => {
+        const result = await carService.generateCar();
+        if(result.success)
```

Comment:
Vi skal også have testet hvad der sker hvis result.success ikke er true
svar fra gpt:
"In principle, test cases should be deterministic and have a clear path of execution. The use of an if statement within the test suggests that there might be different outcomes that are not being explicitly tested, which could lead to flaky tests. Ideally, you want to assert all possible outcomes of the generateCar method. Instead of an if statement, you should have different tests for the success and failure scenarios, where you set up the environment to guarantee each outcome."

Reply:
Helt enig, det lidt noget lort med errorHandler, Der burde aldrig være if statements i test, men hvis ikke der er en if. kan jeg ikke få fat i result.data, så ved ikke hvordan jeg lige skal løse det? Kan vi ændre det i en PR?

Reply:
Mock er nok svaret